### PR TITLE
Add Feature Flag page with new FQDN setting

### PR DESCRIPTION
**Pull request by kilfoyle**
This updates the Standalone Elastic Agent section with a `Feature flag` page, including the new FQDN setting. 

Note: Hold off on merging until https://github.com/elastic/elastic-agent/pull/2218 is merged.

See: [preview](https://observability-docs_2817.docs-preview.app.elstc.co/guide/en/fleet/master/elastic-agent-standalone-feature-flags.html)

Rel: [#1641](https://github.com/elastic/ingest-dev/issues/1641)

### CODE REVIEW

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc
Comment (ycombinator) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc:10]: > ... exposing any breaking changes that might be inconsistent with the naming formats expected in your configured {agent} output.

To future-proof this introduction, I wonder if we should generalize it a bit more.  Besides exposing breaking changes I can also see feature flags being useful for exposing experimental features.   Also, I can see feature flags changing behaviors within Agent or it's components (Beats, Endpoint, etc.) without necessarily impacting just the configured output. 

I'll defer to @cmacknz @joshdover @nimarezainia for input here on how broadly it would make sense to document the purpose of feature flags here.

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc
Comment (ycombinator) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc:29]: Missing `the` before `host.name`?

```suggestion
When enabled, information provided about the current host through the <<host-provider,host.name>> key is in FQDN format (`somehost.example.com` rather than `somehost`). This helps you to distinguish between hosts on different domains that have similar names. With `fqdn` enabled, the fully qualified hostname allows each host to be more easily identified when viewed in {kib}.
```

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc
Comment (kilfoyle) [reply docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc:10]: Definitely! How about something like this?
> The Feature Flags section of the `elastic-agent.yml` config file contains settings in {agent} that are disabled by default. These may include experimental features, changes to behaviors within {agent} or it's components, or settings that could cause a breaking change. For example a setting that changes information included in events might be inconsistent with the naming pattern expected in your configured {agent} output.

Comment (ycombinator) [reply docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc:10]: That looks great, @kilfoyle.  Thanks!

**Comment by ycombinator** [comment]
@kilfoyle A couple of notes tangentially related to this PR:
- We should wait to merge this PR until after https://github.com/elastic/elastic-agent/pull/2218 has been merged.  https://github.com/elastic/elastic-agent/pull/2218 is what actually implements support for Feature Flags in Elastic Agent.
- Do you plan to make an analogous PR for documenting Feature Flags support in Beats? The configuration for Beats is almost the same as it is for Elastic Agent; just that instead of `agent.features:`, you'll have `features:`.  See also https://github.com/elastic/beats/pull/34456.

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc
Comment (kilfoyle) [reply docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc]: Changed as suggested above, via [4d95589](https://github.com/elastic/observability-docs/pull/2817/commits/4d955892fee2f8fcd6b8ca8ddb209be57bb3216e).

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc
Comment (ycombinator) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc:38]: I think somewhere within this section on FQDN, we need to address https://github.com/elastic/ingest-dev/issues/1641#issuecomment-1482574954, specifically the bit about pre-requisites for FQDN reporting to work as expected after users enable the FQDN feature flag.  

For FQDN reporting to work as expected, the hostname of the current host must either:
- have a CNAME entry defined in DNS, or
- have one of its corresponding IP addresses respond successfully to a reverse DNS lookup.

If neither of these pre-requisites are satisfied, `host.name` will continue to report the hostname of the current host as if the FQDN feature flag were not enabled.

Comment (kilfoyle) [reply docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc:45]: Thanks for catching that! I've added the prereqs, as shown:

Screenshot 2023-04-03 at 11 29 02 AM

Comment (ycombinator) [reply docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc:47]: > If neither of these pre-requisites are satisfied, `host.name` will continue to report the hostname of the current host as if the FQDN feature flag were not enabled.

Do you think it's worth mentioning this "fallback" behavior as well or would that be redundant information for users?  It's certainly not _actionable_ information for users but it might explain why they're still seeing the host's hostname despite enabling the FQDN feature flag.  WDYT?

Comment (ycombinator) [reply docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc:47]: @kilfoyle Looks like we crossed wires there 🙂.  I see you already incorporated my last comment in the doc.  Thanks!

Comment (kilfoyle) [reply docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc:47]: I missed that part by accident on the first round. 😄

**Review by ycombinator — APPROVED** [review]
Preview LGTM.  Thanks @kilfoyle for taking care of this!

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc
Comment (karenzone) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-standalone-features.asciidoc:10]:
```suggestion
The elastic-agent.yml config file contains `Feature Flags` settings that are disabled by default. These may include experimental features, changes to behaviors within {agent} or its components, or settings that could cause a breaking change. For example a setting that changes information included in events might be inconsistent with the naming pattern expected in your configured {agent} output.
```

**Comment by ycombinator** [comment]
> Note: Hold off on merging until [elastic/elastic-agent#2218](https://github.com/elastic/elastic-agent/pull/2218) is merged.

@kilfoyle https://github.com/elastic/elastic-agent/pull/2218 has now been merged, so you can go ahead with merging this PR here.  Thanks!

**Pull request merged**
